### PR TITLE
Fix CPU offload + disk offload tests

### PR DESCRIPTION
**Pull request by LysandreJik**
Passing to safetensors serialization by default highlighted a few issues that we have with safetensors.

This PR fixes the issue, which is principally linked to weight sharing.

### CODE REVIEW

**Comment by LysandreJik** [comment]
@amyeroberts @patrickvonplaten if you feel uneasy with merging this right before the release, I'm fine with reverting the safetensors serialization by default to let it sit on `main` for a while longer. The release is going to be very packed already so it's fine for me.

**Review by amyeroberts — APPROVED** [review]
Thanks for finding the fix so quickly!

**Comment by amyeroberts** [comment]
@LysandreJik The change LGTM and seems to address some underlying issues. Re default safetensors serialization, I'm happy for it to be part of this release as long as some of the slow tests on the most popular models (bert, llama, wav2vec2, whisper, clip etc.) are good.

File: src/transformers/models/bart/modeling_bart.py [hunk 1]
```diff
@@ -1125,6 +1125,11 @@ def __init__(self, config: BartConfig):
         # Initialize weights and apply final processing
         self.post_init()
 
+    def _tie_weights(self):
```
Comment (patrickvonplaten) [inline src/transformers/models/bart/modeling_bart.py:1128]: Ah nice, noticed this as well actually :sweat_smile: 
https://github.com/huggingface/transformers/pull/27203/files#r1378948159

File: src/transformers/modeling_utils.py [hunk 2]
```diff
@@ -4520,7 +4520,9 @@ def expand_device_map(device_map, param_names):
     """
     new_device_map = {}
     for module, device in device_map.items():
-        new_device_map.update({p: device for p in param_names if p == module or p.startswith(f"{module}.")})
+        new_device_map.update(
```
Comment (patrickvonplaten) [inline src/transformers/modeling_utils.py:4523]: Nice catch!

**Review by patrickvonplaten — APPROVED** [review]
Clean!

**Comment by LysandreJik** [comment]
Thanks both for your reviews! I'll go ahead and merge this, sorry but you'll have the conflict Patrick :grin:

**Pull request merged**